clueless/static/js/controllers.js

```diff
@@ -200,9 +207,23 @@ class GameHub {
         this.gameState.players[i].character.position = message.location;
       }
     }
+    // I don't think a toast is needed for this since you can see the move happen, but we could always add one.
   }
   handleMsgSuggestion(message) {
-    this.gamePanel.showToast('Suggestion: ' + JSON.stringify(message));
+    // Get which player has won by character id
+    var player;
+    for (var i = 0; i < gameHub.gameState.players.length; i++) {
+      if (gameHub.gameState.players[i].character.id == message.player) {
+        player = gameHub.gameState.players[i];
+      }
+    }
```

Comment:
There's a lot of this throughout different messages. Is `gameHub.gameState.players` not sorted? Just sorting it after it's started avoids having to do this over and over; then it's just `var player = gameHub.gameState.players[message.player];`

Reply:
Agreed, this could use a refactor. Good idea on the sort, I will look into it.

Reply:
There was a bug in PlayerTurn too, this refactor will fix it, so good call!